crowbar_framework/lib/upgrade.rb

```diff
@@ -0,0 +1,62 @@
+class Upgrade
```

Comment:
can you add an `attr_accessor :upgrade_progress` here? then you also don't need the `set_progress` setter anymore.

and otherwise I think you can only access the `upgrade_progress` hash through `Upgrade.new.as_json` which might look a bit inconsistent in the controller